sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/context/AuthTokenContext.java

```diff
@@ -352,16 +358,25 @@ public synchronized void removeOldestTokenIfMatching(String token) {
      * @param authTokenInfo auth token to check for expiration.
      */
     public void checkIfTokenNeedsToBeRefreshed(@NonNull AuthTokenInfo authTokenInfo) {
+        RefreshListener refreshListener = mRefreshListener.get();
+        if (refreshListener != null) {
+            String accountId = getAccountIdToRefreshAndUpdateCache(authTokenInfo);
+            if (accountId != null) {
+                refreshListener.onTokenRequiresRefresh(accountId);
+            }
+        }
+    }
+
+    private synchronized String getAccountIdToRefreshAndUpdateCache(@NonNull AuthTokenInfo authTokenInfo) {
```

Comment:
Any reason of extracting this as a method? I would rather combine two methods to improve readability.
`getAccountIdToRefreshAndUpdateCache` method doesn't need to return a value (makes more sense to return boolean instead of String), the method is more likely detecting a token that is required to be update so returning null for accountId would mislead the behavior of the method.

Reply:
It's for synchronized, need to be at method level for code coverage.